plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java

```diff
@@ -42,83 +47,124 @@
+        MLFeatureEnabledSetting mlFeatureEnabledSetting
     ) {
         super(MLModelGroupGetAction.NAME, transportService, actionFilters, MLModelGroupGetRequest::new);
         this.client = client;
+        this.sdkClient = sdkClient;
         this.xContentRegistry = xContentRegistry;
         this.clusterService = clusterService;
         this.modelAccessControlHelper = modelAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLModelGroupGetResponse> actionListener) {
         MLModelGroupGetRequest mlModelGroupGetRequest = MLModelGroupGetRequest.fromActionRequest(request);
         String modelGroupId = mlModelGroupGetRequest.getModelGroupId();
-        GetRequest getRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+        String tenantId = mlModelGroupGetRequest.getTenantId();
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, actionListener)) {
+            return;
+        }
+
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = new GetDataObjectRequest.Builder()
+            .index(ML_MODEL_GROUP_INDEX)
+            .id(modelGroupId)
+            .tenantId(tenantId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
+
         User user = RestActionUtils.getUserContext(client);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<MLModelGroupGetResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            client.get(getRequest, ActionListener.wrap(r -> {
-                if (r != null && r.isExists()) {
-                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
-                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            ActionListener<MLModelGroupGetResponse> wrappedListener = ActionListener.runBefore(actionListener, context::restore);
 
-                        MLModelGroup mlModelGroup = MLModelGroup.parse(parser);
-                        modelAccessControlHelper.validateModelGroupAccess(user, modelGroupId, client, ActionListener.wrap(access -> {
-                            if (!access) {
-                                wrappedListener
-                                    .onFailure(
-                                        new OpenSearchStatusException(
-                                            "User doesn't have privilege to perform this operation on this model group",
-                                            RestStatus.FORBIDDEN
-                                        )
-                                    );
+            sdkClient
```

Comment:
High level comment: This code block have multiple nested if/else block. Better to refactor and break into smaller functions.

Reply:
@arjunkumargiri in general, true.

In practice, as I've been migrating `client.foo()` calls to `sdkClient.fooDataObject()` calls I have attempted to preserve the existing content, messy nesting and all.  My goal is migrating, not refactoring for efficiency, even if that's generally a good thing.   

My philosopy here is that the diff is much more easily reviewed by reviewers if the only change is between ActionListener and CompletionStage.  More precisely, the pattern:
```java
client.crud(crudRequest, ActionListener.wrap(
    r -> {
        // foo
    }, e -> {
        // bar
    }));
``` 
can be replaced by
```java
sdkClient.crudAsync(crudDataObjectRequest, executor).whenComplete({
    if (throwable == null) {
        // foo
    } else {
        // extract Exception from throwable
        // bar
    }
});
```

Reply:
See for example [this commit](https://github.com/opensearch-project/ml-commons/commit/a4ca16bc705e9d90f72debd203fe12ca60ace392).  This is really just applying the above pattern and updating the tests.